src/history.js

```diff
@@ -214,9 +215,13 @@ class History extends Emitter {
     this.head = action
     this.size += 1
 
+    action.on('change', this.reconcile, this)
+
     this._emit('append', action)
 
-    action.on('change', this.reconcile, this)
+    if (status && action.command !== START) {
+      this.reconcile(action)
+    }
```

Comment:
Fun little bug. if you did `repo.append(action, 'status')`, reconciliation wouldn't happen the first time!